packages/cdk-assets/lib/private/docker.ts

```diff
@@ -37,9 +37,9 @@ export class Docker {
       '--tag', options.tag,
       ...options.target ? ['--target', options.target] : [],
       ...options.file ? ['--file', options.file] : [],
-      options.directory,
+      '.'
     ];
-    await this.execute(buildCommand);
+    await this.execute(buildCommand, { cwd: options.directory });
```

Comment:
lgtm fwiw.  The below alternative patch also worked for me if for some reason the reviewers wanted to avoid setting cwd (I don't know why that would be an issue):

```patch
--- cdk-assets/lib/private/handlers/container-images.js.orig    2020-03-24 06:03:56.670862836 +0000
+++ cdk-assets/lib/private/handlers/container-images.js 2020-03-24 05:58:01.971071483 +0000
@@ -51,7 +51,7 @@
             tag: this.localTagName,
             buildArgs: source.dockerBuildArgs,
             target: source.dockerBuildTarget,
-            file: source.dockerFile,
+            file: path.resolve(fullPath, source.dockerFile),
         });
     }
 }

```